src/main.rs

```diff
@@ -1,112 +1,176 @@
+};
 
 /// Command line tool for C projects
 #[derive(Parser, Debug, Clone)]
 #[command(version, about, long_about = None)]
 struct Args {
     /// Name of the project
-    project_name: String,
+    // project_name: Option<String>,
+    #[command(subcommand)]
+    command: Option<Commands>,
+}
 
-    /// Create lib directory whith files named <FILE>.c and <FILE>.h
-    #[arg(long, value_name = "FILE")]
-    lib: Option<String>,
+#[derive(Subcommand, Debug, Clone)]
+enum Commands {
+    /// Create new C Proejct with name
+    CNew {
+        project_name: String,
+
+        /// Create lib directory whith files named <FILE>.c and <FILE>.h
+        #[arg(long, value_name = "FILE")]
+        lib: Option<String>,
+    },
+
+    /// Build ./src/*.c files and moved to ./bin/
+    BuildC,
+    /// Build and run project
+    BulidCRunC,
 }
 
 #[derive(Clone)]
 struct Cinit(Args);
 
 impl Cinit {
     fn new() -> Self {
-        let cinit = Cinit(Args::parse());
-        let _ = fs::create_dir(cinit.clone().0.project_name);
-
-        cinit
+        Cinit(Args::parse())
     }
+}
+
+fn c_new(project_name: &String, lib: Option<String>) -> Result<(), Box<dyn std::error::Error>> {
+    create_sub_directory(project_name, "src")?;
 
-    fn create_sub_directory(&self, sub_dir: &str) -> Result<(), std::io::Error> {
-        let full_path = format!("./{}/{}", &self.0.project_name, &sub_dir);
-        fs::create_dir_all(&full_path)?;
-        Ok(())
+    if let Some(lib_file_name) = lib {
+        create_lib_folder(project_name, &lib_file_name)?;
     }
 
-    fn create_file(&self, sub_dir: &str, name: &str) -> Result<(), Box<dyn std::error::Error>> {
-        let file_dir = if sub_dir.is_empty() {
-            format!("./{}/{}", self.0.project_name, name)
-        } else {
-            format!("./{}/{}/{}", self.0.project_name, sub_dir, name)
-        };
-
-        File::create(&file_dir)?;
-
-        // Writting deafult code
-        if name == "main.c" {
-            self.write_main_c(&file_dir);
-        } else if name == "Makefile" {
-            self.write_makefile(&file_dir);
-        } else if name == "README.md" {
-            self.write_readme(&file_dir);
-        } else {
-            self.write_default_content(&file_dir);
-        }
+    create_file(project_name, "src", "main.c")?;
+    create_file(project_name, "", "README.md")?;
+    create_file(project_name, "", "Makefile")?;
 
-        Ok(())
-    }
+    Ok(())
+}
 
-    fn write_main_c(&self, file_dir: &str) {
-        let content = "#include <stdio.h>\n\nint main() {\n    printf(\"Hello, world!\\n\");\n    return 0;\n}\n";
-        let _ = fs::write(file_dir, content);
-        println!("Content written to {}", file_dir);
-    }
+fn build_c() -> Result<(), std::io::Error> {
+    // Ejecutar GCC para compilar el programa
+    let gcc_status = Command::new("gcc")
+        .args(["src/main.c", "-Wall", "-o", "main"])
+        .status()
+        .expect("Failed to execute GCC");
 
```

Comment:
I think this command should be using the build system selected by the user.

If the user selects that wants a Makefile in the project I think this command should execute `make`

And you should consider the possibility of a different build system. For example `nob` , our dear friend tsoding is using it in their projects.

Reply:
For the moment the build command will only compile using gcc. Future versions will accept the build system selected by the user.

Reply:
Put an issue and reference the issue in this conversation then

Reply:
#6